Lib/argparse.py

```diff
@@ -1631,6 +1639,14 @@ def _handle_conflict_resolve(self, action, conflicting_actions):
             if not action.option_strings:
                 action.container._remove_action(action)
 
+    def _check_help(self, action):
+        if action.help and hasattr(self, "_get_formatter"):
+            formatter = self._get_formatter()
+            try:
+                formatter._expand_help(action)
+            except (ValueError, TypeError, KeyError):
```

Comment:
Is it maybe worth splitting these up and making the error more descriptive? In general, I think it's valuable to have error messages be actionable, and I think this could benefit from more specificity.

Reply:
Details are available from a traceback.
```pycon
>>> '%' % {'prog': 'PROG'}
Traceback (most recent call last):
  File "<python-input-0>", line 1, in <module>
    '%' % {'prog': 'PROG'}
    ~~~~^~~~~~~~~~~~~~~~~~
ValueError: incomplete format
>>> '%(' % {'prog': 'PROG'}
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    '%(' % {'prog': 'PROG'}
    ~~~~~^~~~~~~~~~~~~~~~~~
ValueError: incomplete format key
>>> '%(foo)s' % {'prog': 'PROG'}
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    '%(foo)s' % {'prog': 'PROG'}
    ~~~~~~~~~~^~~~~~~~~~~~~~~~~~
KeyError: 'foo'
>>> '%(prog)i' % {'prog': 'PROG'}
Traceback (most recent call last):
  File "<python-input-3>", line 1, in <module>
    '%(prog)i' % {'prog': 'PROG'}
    ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
TypeError: %i format: a real number is required, not str
```
We cannot be more specific. We could even just leave the original exception to pop up. A new ValueError just adds a little context to the original exception.

The problem was that the user did not get any error if they did not test the parser's help. So the bug could be discovered by the end user of the program, not the author of the code. Now the author will get an error and they have all information to fix it.

Reply:
Thanks for sharing that output, that looks good to me.